### PR TITLE
Relative Sort Array in cpp

### PR DESCRIPTION
**Pull request by Debashish-hub**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added sorting an array according to another array in c++ in sorting section and updated readme section also.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#6508

### CODE REVIEW

File: C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+// 2 2 1 1 8 8 3 5 6 7 9 
+
+// Input:
+
+// 11 4
+// 2 1 2 5 7 1 9 3 6 8 8
+// 99 22 444 56
+// Output: 
+
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp]: Time and Space Complexity are missing

File: C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+
+int main(int argc, char *argv[]) 
+{ 
+	
+	int t;
+	cin >> t;
+	while(t--){
+	    int n, m;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp]: Use a good naming convention

File: C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	    cin >> n >> m;
+	    vector<int> a1(n);
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp]: Use a good naming convention

File: C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp (hunk 2 above)
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp]: Display a message in print form as what are you taking input from the user

File: C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	    vector<int> a2(m);
+	    for(int i = 0;i<n;i++){
+	        cin >> a1[i];
+	    }
+	    for(int i = 0;i<m;i++){
+	        cin >> a2[i];
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp]: Display a message in print form as what are you taking input from the user

File: C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	    }
+	    Solution ob;
+	    a1 = ob.sortA1ByA2(a1, n, a2, m); 
+	    for (int i = 0; i < n; i++) 
+		    cout<<a1[i]<<" ";
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/Sorting_an_array_according_to_another_array.cpp]: Display a message in print form as what are you showing output to the user

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Debashish-hub Do the suggested changes

**Comment by Debashish-hub** [comment]
ok, I will do

**Comment by Debashish-hub** [comment]
Changes are done. @siddharth25pandey  Please check it

File: C-Plus-Plus/sort/Relative_Sort_Array.cpp [hunk 6]
```diff
@@ -0,0 +1,111 @@
+//Relative Sort Array (Sorting an array according to the other) 
+
+// Given two integer arrays A1[ ] and A2[ ] of size N and M respectively. Sort the first array A1[ ] such that all the relative 
+// positions of the elements in the first array are the same as the elements in the second array A2[ ].
+
+// Note: If elements are repeated in the second array, consider their first occurance only.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/Relative_Sort_Array.cpp]: Use Multiline comments

File: C-Plus-Plus/sort/Relative_Sort_Array.cpp [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/Relative_Sort_Array.cpp]: remove extra spacing

File: C-Plus-Plus/sort/Relative_Sort_Array.cpp [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    } 
+};
+
+
+int main(int argc, char *argv[]) 
+{ 
+	
+	int t;
+	cin >> t;
+	while(t--){
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/Relative_Sort_Array.cpp]: - Remove test case part as we don't have online judge to test 
- Use proper display message how you are reading and writing outputs

File: C-Plus-Plus/sort/Relative_Sort_Array.cpp [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+	    int n, m;
+	    cin >> n >> m;                 // 11 4
+	    vector<int> a1(n);
+	    vector<int> a2(m);
+	    for(int i = 0;i<n;i++){
+	        cin >> a1[i];             // 2 1 2 5 7 1 9 3 6 8 8
+	    }
+	    for(int i = 0;i<m;i++){
+	        cin >> a2[i];            // 2 1 8 3
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/Relative_Sort_Array.cpp]: remove the redundant comments

File: C-Plus-Plus/sort/Relative_Sort_Array.cpp [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+// 99 22 444 56
+// Output: 
+
+// 1 1 2 2 3 5 6 7 8 8 9
+
+//Time Complexity : 
+// O(m+n)
+
+//Space Complexity:
+//O(n)    if n>m
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/Relative_Sort_Array.cpp]: Remove extra test case and rewrite it using the following format.

```
/*

Sample Input/Output:

Input:

Output:


Time complexity:
Space complexity:

*/

```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/sort/Relative_Sort_Array.cpp (hunk 10 above)
Comment (ankitaggarwal23) [reply C-Plus-Plus/sort/Relative_Sort_Array.cpp]: one sample input is enough

**Comment by Debashish-hub** [comment]
Please check @ankitaggarwal23

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**